pytorch_lightning/utilities/types.py

```diff
@@ -0,0 +1,8 @@
+from typing import Dict, Iterator, Optional, Union
+
+import torch
+from torchmetrics import Metric
+
+_METRIC = Union[Metric, torch.Tensor, int, float]
```

Comment:
is it necessary to put the underscore?

Reply:
Do we want users to import these and care about backwards compatibility?

Reply:
yes, as it is protected - just internal usage
BUT for step types, it would make sense to make it public so users can reuse our typing
cc: @ananthsub